pkg/apis/controlplane/types.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controlplane
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// Address of a cluster
+type ClusterAddress struct {
+	// URL to access the cluster
+	Url string `json:"url"`
```

Comment:
At other places in the API we refer to it as Address (can be a hostname, IP address or IP address and port).

Also, we might need both an internal and external address.
You can look at ServerAddressByClientCIDR that we use at another place: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/unversioned/types.go#L347